imagebuildah/stage_executor.go

```diff
@@ -362,6 +364,34 @@ func (s *StageExecutor) Copy(excludes []string, copies ...imagebuilder.Copy) err
 		stripSetgid := false
 		preserveOwnership := false
 		contextDir := s.executor.contextDir
+
+		// If we are copying files via heredoc syntax, then
+		// its time to create these temporary files on host
+		// and copy these to container
+		if len(copy.Files) > 0 {
+			copySources := []string{}
+			for _, file := range copy.Files {
+				data := file.Data
+				// remove first break line added while parsing heredoc
+				data = strings.TrimPrefix(data, "\n")
+				// add breakline when heredoc ends for docker compat
+				data = data + "\n"
+				tmpFile, err := os.Create(filepath.Join("/var/tmp/", filepath.Base(filepath.ToSlash(file.Name))))
```

Comment:
Use `parse.GetTempDir()` here?

Reply:
`os.Create()` is affected by the current process umask.  There needs to be a `tmpFile.Chmod()` in here somewhere.